docs/developer-guide/Tests/ipsec-auth-e2e-test.md

```diff
@@ -0,0 +1,261 @@
+# IPSec & Offload Authorization E2E Test Guide
+
+This document provides a step-by-step guide for executing the IPSec and Offload Authorization E2E tests for Kmesh. These tests ensure the reliability, security, and functionality of the IPSec feature and the Offload Authorization mechanisms.
+
+## Prerequisites
+
+Before running the tests, ensure the following:
+
+- **Kubernetes Cluster**: A two-node Kubernetes cluster with Kmesh installed.
+- **Tools**: `kubectl`, `tcpdump`, and `kmeshctl`.
+- **Applications**: `httpbin` and `sleep` applications deployed in the cluster.
+
+## IPSec E2E Tests
+
+### 1. Basic Connectivity Test
+
+This test verifies the establishment of IPSec tunnels and the correctness of encrypted communication.
+
+#### Steps
+
+1. Deploy the `httpbin` and `sleep` applications on different nodes:
+
+   ```bash
+   kubectl apply -f httpbin.yaml
+   kubectl apply -f sleep.yaml
+   ```
+
+2. Verify connectivity between the applications:
+
+   ```bash
+   kubectl exec <sleep-pod> -- curl http://<httpbin-service>
+   ```
+
+3. Check IPSec state and policy rules:
+
+   ```bash
+   ip xfrm state show
+   ip xfrm policy show
```

Comment:
Should tell user what output is correct. The same applies to all subsequent command executions.